### PR TITLE
Fix potential naming clash when writing traces with tensorboard_trace_handler

### LINKED ISSUE #82915: DISABLED test_tensorboard_trace_handler (__main__.TestProfiler)
Platforms: mac, macos

This test was disabled because it is failing in CI. See [recent examples](https://hud.pytorch.org/flakytest?name=test_tensorboard_trace_handler&suite=TestProfiler&file=test_profiler.py) and the most recent trunk [workflow logs](https://github.com/pytorch/pytorch/runs/7695478091).

Over the past 3 hours, it has been determined flaky in 1 workflow(s) with 1 red and 1 green.

cc @robieta @chaekit @aaronenyeshi @ngimel @nbcsm @guotuofeng @guyang3532 @gaoteng-git @tiffzhaofb

### PR DESCRIPTION
**Pull request by huydhn**
Fixes https://github.com/pytorch/pytorch/issues/82915

This rare flaky issue caught my attention today when it failed flakily on MacOS in https://github.com/pytorch/pytorch/actions/runs/4494182574/jobs/7906827531.  The test expected 3 traces to be written but got only 2 of them.

Looking a bit closer into the `tensorboard_trace_handler` function, it looks like there is a potential filename clash here.  The millisecond since epoch `"{}.{}.pt.trace.json".format(worker_name, int(time.time() * 1000))` is used as part of the name.  As `tensorboard_trace_handler` is used as a callback handle in the test, the names look too close to each other (1-millisecond apart), i.e.

```
huydo-mbp_13494.1679526197252.pt.trace.json
huydo-mbp_13494.1679526197253.pt.trace.json
huydo-mbp_13494.1679526197250.pt.trace.json
```

Switching to nanosecond reduces the chance of two or more of them having the same timestamp while keeping the naming convention intact, i.e. `huydo-mbp_13804.1679526325182878000.pt.trace.json`

I suspect that this is also the cause of Windows flakiness.

cc @robieta @chaekit @aaronenyeshi @ngimel @nbcsm @guotuofeng @guyang3532 @gaoteng-git @tiffzhaofb @dzhulgakov @davidberard98

### CODE REVIEW

File: torch/profiler/profiler.py [hunk 1]
```diff
@@ -326,7 +326,8 @@ def handler_fn(prof) -> None:
                 raise RuntimeError("Can't create directory: " + dir_name) from e
         if not worker_name:
             worker_name = "{}_{}".format(socket.gethostname(), str(os.getpid()))
-        file_name = "{}.{}.pt.trace.json".format(worker_name, int(time.time() * 1000))
+        # Use nanosecond here to avoid naming clash when exporting the trace
+        file_name = "{}.{}.pt.trace.json".format(worker_name, time.time_ns())
```
Comment (malfet) [inline torch/profiler/profiler.py:330]: We live in Python-3 world now, why not use f-strings?
```suggestion
        file_name = f"{worker_name}.{time.time_ns()}.pt.trace.json"
```

**Review by malfet — APPROVED** [review]

**Review by aaronenyeshi — APPROVED** [review]
Looks great! Thank you for fixing this! Please fix Nikita's suggestions, and feel free to merge

**Pull request closed without merging**